pkg/timeline/adaptor/repository/dummyCache.ts

```diff
@@ -0,0 +1,57 @@
+import { Result } from '@mikuroxina/mini-fn';
+
+import type { AccountID } from '../../../accounts/model/account.js';
+import { type Note, type NoteID } from '../../../notes/model/note.js';
+import type {
+  CacheObjectKey,
+  TimelineNotesCacheRepository,
+} from '../../model/repository.js';
+
+export class InMemoryTimelineCacheRepository
+  implements TimelineNotesCacheRepository
+{
+  private readonly data: Map<CacheObjectKey, NoteID[]>;
+  constructor(data: [AccountID, NoteID[]][] = []) {
+    this.data = new Map(
+      data.map(([accountID, noteIDs]) => [
+        `timeline:home:${accountID}`,
+        noteIDs,
+      ]),
+    );
+  }
+
+  private generateObjectKey(accountID: AccountID): CacheObjectKey {
+    return `timeline:home:${accountID}`;
+  }
+
+  async addNotesToHomeTimeline(
+    accountID: AccountID,
+    notes: Note[],
+  ): Promise<Result.Result<Error, void>> {
+    if (!this.data.has(this.generateObjectKey(accountID))) {
+      this.data.set(
+        this.generateObjectKey(accountID),
+        notes.map((note) => note.getID()),
+      );
+      return Result.ok(undefined);
+    }
+    const fetched = this.data.get(this.generateObjectKey(accountID))!;
+    // NOTE: replace by updated object
+    this.data.delete(this.generateObjectKey(accountID));
+
+    fetched.push(...notes.map((note) => note.getID()));
+    this.data.set(this.generateObjectKey(accountID), fetched);
```

Comment:
```suggestion
    const objectKey = this.generateObjectKey(accountID);
    if (!this.data.has(objectKey)) {
      this.data.set(
        objectKey,
        notes.map((note) => note.getID()),
      );
      return Result.ok(undefined);
    }
    const fetched = this.data.get(objectKey)!;
    // NOTE: replace by updated object
    this.data.delete(objectKey);

    fetched.push(...notes.map((note) => note.getID()));
    this.data.set(objectKey, fetched);
```
何回もキーを作成しているところは, 変数で 1 つにまとめられそうです.